src/decode.rs

```diff
@@ -275,3 +284,16 @@ fn decode_numeric(esc: &str, radix: u32) -> Result<char, DecodeErrKind> {
         Err(..) => Err(MalformedNumEscape),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    // Regression test for "out of bounds" panic in sloppy decoder.
+    #[test]
+    fn test_decode_panic() {
+        let bytes = b"&#1 ;foobar";
```

Comment:
Did the same panic happen with a whitespace?

Reply:
Yes, just as newline it's invalid and clears `buf`.